packages/components/src/components/hds/application-state/index.ts

```diff
@@ -22,7 +31,30 @@ export interface HdsApplicationStateSignature {
   Element: HTMLDivElement;
 }
 
-const HdsApplicationStateComponent =
-  TemplateOnlyComponent<HdsApplicationStateSignature>();
+export default class HdsApplicationStateComponent extends Component<HdsApplicationStateSignature> {
+  constructor(owner: unknown, args: HdsApplicationStateSignature['Args']) {
+    super(owner, args);
+
+    const validAlignValues: HdsApplicationStateAligns[] = Object.values(
+      HdsApplicationStateAlignValues
+    );
 
-export default HdsApplicationStateComponent;
+    if (
+      this.args.align != null &&
+      !validAlignValues.includes(this.args.align)
+    ) {
+      throw new Error(`Invalid align value: ${this.args.align}`);
+    }
+  }
+
+  get align(): HdsApplicationStateAligns {
+    return this.args.align ?? HdsApplicationStateAlignValues.Left;
+  }
+
+  get classNames(): string {
+    return [
+      'hds-application-state',
+      `hds-application-state--align-${this.align}`,
+    ].join(' ');
+  }
```

Comment:
I know that at first sight sense strange how we do it in other components, but can you stick to the standard/pattern of how we do it in other components? there are good reasons for it (I can explain, ig you want)

https://github.com/hashicorp/design-system/blob/main/packages/components/src/components/hds/accordion/index.ts#L81-L91